### PR TITLE
Enable scaled_mm on sm89 devices

### LINKED ISSUE #118703: RuntimeError: torch._scaled_mm is only supported on devices with compute capability >= 9.0)
### 🚀 The feature, motivation and pitch

Possible to add support for the NVIDIA Ada Lovelace cards, which have a compute capability of 8.9?

NVIDIA's TransformerEngine works, so the feature is supported in the hardware.

Thanks!

@drisspg 

### Alternatives

_No response_

### Additional context

_No response_

cc @yanbing-j @vkuzo @albanD @kadeng

### PR DESCRIPTION
**Pull request by drisspg**
Fixes #118703

cc @ptrblck @yanbing-j @vkuzo @albanD @kadeng

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM

**Comment by ptrblck** [comment]
Let me build and test the PR tomorrow.

File: test/test_matmul_cuda.py [hunk 1]
```diff
@@ -204,10 +204,19 @@ def _expand_to_batch(t: torch.Tensor):
         self.assertEqual(out1_gpu, out2_gpu[0])
 
 
+def scaled_mm_supported_device():
+    return torch.cuda.is_available() and (
+        torch.cuda.get_device_capability >= (9, 0) or torch.cuda.get_device_capability() == (8, 9)
```
Comment (ptrblck) [inline test/test_matmul_cuda.py:209]: typo:
```python
torch.cuda.get_device_capability >= (9, 0)
```
->
```python
torch.cuda.get_device_capability() >= (9, 0)
```

File: test/test_matmul_cuda.py [hunk 2]
```diff
@@ -301,11 +308,11 @@ def test_error_message_fp8_non_h100(self, device) -> None:
         y = torch.rand((m, l), device=device).to(torch.float8_e4m3fn).t()
         self.assertRaisesRegex(
             RuntimeError,
-            r"torch\.\_scaled\_mm is only supported on devices with compute capability \>\= 9\.0",
+            r"torch\.\_scaled\_mm is only supported on devices with compute capability \>\= 9\.0 or 8\.9",
```
Comment (ptrblck) [inline test/test_matmul_cuda.py:311]: L303 and L304 should be changed to:
```
    @unittest.skipIf(scaled_mm_supported_device(),
                     "This test is only for devices with compute capability < 8.9")
```
The test name could also be changed to:
```python
test_error_message_fp8_pre_sm89
```

**Comment by drisspg** [comment]
A

> After the changes mentioned in my review the PR passes on an L4 (sm_89):
> 
> ```python
> python test_matmul_cuda.py -v 
> test_error_message_fp8_non_h100_cuda (__main__.TestFP8MatmulCudaCUDA) ... skipped 'This test is only for devices with compute capability < 8.9'
> test_float32_output_errors_with_bias_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_float8_basics_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_float8_bias_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_float8_bias_relu_edgecase_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_float8_scale_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_float8_scale_fast_accum_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_non_divisible_leading_dim_bias_False_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_non_divisible_leading_dim_bias_True_cuda (__main__.TestFP8MatmulCudaCUDA) ... ok
> test_cublas_addmm_alignment_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_reduced_precision_size_10000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_reduced_precision_size_10000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_reduced_precision_size_1000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_reduced_precision_size_1000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_reduced_precision_size_100_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_reduced_precision_size_100_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_10000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_10000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_10000_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_1000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_1000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_1000_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_100_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_100_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_addmm_size_100_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_1_10000_10000_10000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_1_10000_10000_10000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_1_10000_10000_10000_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_1_10000_1000_10000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_1_10000_1000_10000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_1_10000_1000_10000_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_2_1000_1000_1000_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_2_1000_1000_1000_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_2_1000_1000_1000_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_2_100_100_100_cuda_bfloat16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_2_100_100_100_cuda_float16 (__main__.TestMatmulCudaCUDA) ... ok
> test_cublas_baddbmm_large_input_2_100_100_100_cuda_float32 (__main__.TestMatmulCudaCUDA) ... ok
> test_mixed_dtypes_linear_cuda_bfloat16 (__main__.TestMixedDtypesLinearCudaCUDA) ... ok
> test_mixed_dtypes_linear_cuda_float16 (__main__.TestMixedDtypesLinearCudaCUDA) ... ok
> 
> ----------------------------------------------------------------------
> Ran 39 tests in 39.303s
> 
> OK (skipped=1)
> ```

File: test/test_matmul_cuda.py (hunk 1 above)
Comment (malfet) [reply test/test_matmul_cuda.py:209]: Hmm, how it was not caught in the testing?  As comparing function to a tuple should raise type error
```
>>> range >= (2, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'type' and 'tuple'
```

Comment (drisspg) [reply test/test_matmul_cuda.py]: For provenance: CI tests were failing

**Pull request closed without merging**